### PR TITLE
Test importing passwords from Chrome

### PR DESCRIPTION
**Pull request by garrettr**
Unit tests for #77. Also refactors the importer unit tests to reduce code duplication.

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Test Plan:

`yarn test brave_unit_tests`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

**Comment by bbondy** [comment]
lgtm but @darkdh is the better reviewer, so I'll defer to him to approve or ask for changes.

File: utility/importer/chrome_importer_unittest.cc [hunk 1]
```diff
@@ -123,3 +133,29 @@ TEST(ChromeImporterTest, ImportFavicons) {
...
+      .WillOnce(::testing::SaveArg<0>(&password));
+  EXPECT_CALL(*bridge_, NotifyItemEnded(importer::PASSWORDS));
+  EXPECT_CALL(*bridge_, NotifyEnded());
+
+  importer_->StartImport(profile_, importer::PASSWORDS, bridge_.get());
+
+  EXPECT_EQ("test@example.com",
+            UTF16ToASCII(password.username_value));
+  EXPECT_EQ("testing123",
+            UTF16ToASCII(password.password_value));
```
Comment (darkdh) [inline utility/importer/chrome_importer_unittest.cc]: also need to test the case of `Never Saved List`

**Comment by bbondy** [comment]
Should we close this PR?

**Comment by darkdh** [comment]
Waiting for test case update and I will approve

**Comment by garrettr** [comment]
I will provide an updated test case this week.

**Comment by garrettr** [comment]
This PR now tests importing both "autofillable" and "blacklisted" logins from Chrome, as requested by @darkdh in his review. A login is autofillable or blacklisted depending on whether you click "Yes" or "Never" when Chrome prompts you to save a password.

In addition, I improved the process for generating the test copy of Chrome's `Login Data` database, and documented it in detail.

File: utility/importer/chrome_importer_unittest.cc [hunk 2]
```diff
@@ -140,21 +140,28 @@ TEST_F(ChromeImporterTest, ImportPasswords) {
...
   EXPECT_CALL(*bridge_, NotifyItemEnded(importer::PASSWORDS));
   EXPECT_CALL(*bridge_, NotifyEnded());
 
   importer_->StartImport(profile_, importer::PASSWORDS, bridge_.get());
 
-  EXPECT_EQ("test@example.com",
-            UTF16ToASCII(password.username_value));
-  EXPECT_EQ("testing123",
-            UTF16ToASCII(password.password_value));
+  EXPECT_FALSE(autofillable_login.blacklisted_by_user);
```
Comment (darkdh) [inline utility/importer/chrome_importer_unittest.cc]: can we also verify `signon_realm` for both `autofillable_login` and `blacklisted_login`?

**Pull request merged**